active_plugins/rundeepprofiler.py

```diff
@@ -0,0 +1,375 @@
+            "Model directory", allow_metadata=False, doc="""\
+ Where the model is located.
+
+{IO_FOLDER_CHOICE_HELP_TEXT}
+""".format(**{
+                "IO_FOLDER_CHOICE_HELP_TEXT": _help.IO_FOLDER_CHOICE_HELP_TEXT
+            }))
+
+        def set_directory_fn_executable(path):
+            dir_choice, custom_path = self.model_directory.get_parts_from_path(path)
+            self.model_directory.join_parts(dir_choice, custom_path)
+
+        self.model_filename = Filename(
+            "Model file", "Cell_Painting_CNN_v1.hdf5", doc="Select your DeepProfiler model. ",
+            get_directory_fn=self.model_directory.get_absolute_path,
+            set_directory_fn=set_directory_fn_executable,
+            browse_msg="Choose executable file"
+        )
+
+        self.config_directory = Directory(
+            "Configuration directory", allow_metadata=False, doc="""\
+Select the folder containing the DeepProfiler configuration file.
+
+{IO_FOLDER_CHOICE_HELP_TEXT}
+""".format(**{
+                "IO_FOLDER_CHOICE_HELP_TEXT": _help.IO_FOLDER_CHOICE_HELP_TEXT
+            }))
+
+        def set_directory_config_executable(path):
+            dir_choice, custom_path = self.config_directory.get_parts_from_path(path)
+            self.config_directory.join_parts(dir_choice, custom_path)
+
+        self.config_filename = Filename(
+            "Configuration file", "profiling.json", doc="Select your configuration file.",
+            get_directory_fn=self.config_directory.get_absolute_path,
+            set_directory_fn=set_directory_config_executable,
+            browse_msg="Choose executable file"
+        )
+        
+        self.app_directory = Directory(
+            "DeepProfiler directory", doc="""\
+Select the folder containing the DeepProfiler repository that you cloned to your computer.
+
+{fcht}
+""".format(
+                fcht=IO_FOLDER_CHOICE_HELP_TEXT
+            ),
+        )
+
+
+    def settings(self):
+        return [self.images_list, self.input_object_name, self.assay, self.directory, self.app_directory, 
+                self.model_directory, self.model_filename,
+                self.metadata_plate, self.metadata_well, self.metadata_site,
+                self.config_directory, self.config_filename]
+
+    #
+    # CellProfiler calls "run" on each image set in your pipeline.
+    #
+    def run(self, workspace):
+
+        #
+        # Get directories
+        #         
+        out_directory = self.directory.value.split("|")[1]
+        model_dir = self.model_directory.value.split("|")[1]
+        config_dir = self.config_directory.value.split("|")[1]
+        deep_directory = self.app_directory.value.split("|")[1]
+
+        #
+        # Create folders using deepprofiler setup
+        #
+        executable = f"{deep_directory}\deepprofiler"
+        cmd_setup = f"python {executable} --root={out_directory} setup"
+        subprocess.run(cmd_setup, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, text=True)
```

Comment:
Right now, we're not capturing if we run successfully or not. I think something like this would be good
```
        result = subprocess.run(cmd_setup.split(), stdout=subprocess.PIPE, stderr=subprocess.STDOUT, text=True)
        if result.returncode!=0:
            msg = f"The call to DeepProfiler returned an error; you should try to solve this outside of CellProfiler. DeepProfiler output waws as follows: \n {result.stdout}"
            raise RuntimeError(msg)
```